parquet/src/main/scala/magnolify/parquet/ParquetType.scala

```diff
@@ -175,26 +178,40 @@ object ParquetType {
     override def getName: String = "magnolify"
 
     private var recordConsumer: RecordConsumer = null
+    private var conf: Configuration = null
 
     override def init(configuration: Configuration): hadoop.WriteSupport.WriteContext = {
       if (parquetType == null) {
         parquetType = SerializationUtils.fromBase64[ParquetType[T]](configuration.get(WriteTypeKey))
       }
 
-      val schema = Schema.message(parquetType.schema)
+      val schema = Schema.message(parquetType.schema(configuration))
       val metadata = new java.util.HashMap[String, String]()
-      if (parquetType.avroCompat) {
-        // This overrides `WriteSupport#getName`
-        metadata.put(ParquetWriter.OBJECT_MODEL_NAME_PROP, "avro")
```

Comment:
I did drop the behavior of writing `writer.model.name: avro` if AvroCompat is enabled -- I don't think it makes sense, it should still be Magnolify. I can't think of any reason why this would impact downstream readers -- model name shouldn't matter at all when comparing schema compatibility across files. lf anyone can think of a good reason why this change is breaking let me know...